### PR TITLE
[Docs] Add service, client, and test authoring guidance to the package README

### PR DESCRIPTION
**Pull request by ravindu25**
### Purpose

The README covers the concepts behind `Client` and `Listener`, but does not spell out the conventions for actually writing a service, a client, or tests for a service. These are the details that are most often gotten wrong.

### Changes

Appended three sections:

- **Writing HTTP services** - module-level listener declaration, resource-only services, path parameters in the resource path, `@http:Payload` / `@http:Query` / `@http:Header` usage, and preferring concrete return types.
- **Writing HTTP clients** - module-level `final` clients, direct data binding, and when `http:Response` is actually needed.
- **Writing tests for HTTP services** - file structure, the client resource-access call syntax per HTTP method, and response handling for positive and negative cases.
These package READMEs are also surfaced as reference context in IDE and AI-assisted tooling, so keeping usage guidance in the README benefits both readers and tooling.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary

- Updated the package README with clearer authoring guidance for HTTP services, including module-level listeners, resource-only entry points, path-parameter conventions, payload/query/header annotations, and concrete resource return types.
- Strengthened HTTP client guidance with module-level `final` client declarations and correct direct data binding patterns, including when to use `http:Response`.
- Revised HTTP service testing guidance to ensure consistent test structure and correct client-to-resource call syntax across HTTP methods, including corrected DELETE and PATCH examples and stronger emphasis on `clientEp` being declared `final`.
- Improved the README as reference context for IDE and AI-assisted tooling by aligning examples with the documented service/client/testing conventions.

### CODE REVIEW

File: ballerina/README.md [hunk 1]
```diff
@@ -110,3 +110,120 @@ built-in support for basic authentication, JWT authentication, and OAuth2 authen
 In addition to that, supports both the HTTP/1.1 and HTTP2 protocols and connection keep-alive, content 
 chunking, HTTP caching, data compression/decompression, payload binding, HTTP/2 stream concurrency limiting, and
 authorization can be highlighted as the features of a `Service`.
+
+## Writing HTTP services
+
+When authoring an HTTP service, keep the following conventions in mind:
+
+- An HTTP service always needs an `http:Listener` attached to it. Declare the listener at the module level as a variable and reference it in the service declaration (for example, `listener http:Listener ep = check new (8080);`).
+- A service may contain only resource functions.
```
[the remark this answers is not in the document]
Comment (ravindu25) [reply ballerina/README.md]: fixed

File: ballerina/README.md [hunk 2]
```diff
@@ -110,3 +110,120 @@ built-in support for basic authentication, JWT authentication, and OAuth2 authen
...
+        return account;
+    }
+}
+```
+
+## Writing HTTP clients
+
+- Always declare clients at the module level as `final` variables.
+- Use direct data binding to bind the response to a type whenever possible.
+- Use the `http:Response` type as the return type only when you need to access the headers or status code of the response.
```
[the remark this answers is not in the document]
Comment (ravindu25) [reply ballerina/README.md:165]: Added `final` to `clientEp`.

Skipped the other two: the non-final clients are in the existing Overview section, not part of this change. And the mocked-final-client pattern is for mocking dependencies — `clientEp` is just the client calling the service under test.

File: ballerina/README.md [hunk 3]
```diff
@@ -110,3 +110,120 @@ built-in support for basic authentication, JWT authentication, and OAuth2 authen
...
+    - DELETE - `clientEp->/books.delete(isbn = value);`.
+    - PATCH - `clientEp->/books/[isbn].patch(payload, name = value);`.
+- For non-annotated resource parameters, records are treated as body parameters and other types as query parameters.
+- For negative test cases, cover the scenarios explicitly handled in the service rather than theoretical edge cases.
+
+**Response handling**
+
+- Use direct data binding for positive test cases.
+- For negative test cases, use `http:Response` variables and assert the status codes.
+- Always assign responses to variables with specific types.
```
[the remark this answers is not in the document]
Comment (ravindu25) [reply ballerina/README.md:228]: fixed

**Comment by TharmiganK** [comment]
@ravindu25 May I know the intention of this PR? 

The package README.md can carry the common usage patterns. But IMO having specific guidelines like these are subjective. We could mention it as best practices but I prefer to have it as a different document. Also the tone of the changes seems to be for an AI agent rather than a human
